src/Symfony/Component/Routing/Tests/Matcher/TraceableUrlMatcherTest.php

```diff
@@ -98,4 +99,23 @@ public function getLevels($traces)
 
         return $levels;
     }
+
+    public function testRoutesWithConditions()
+    {
+        $routes = new RouteCollection();
+        $routes->add('foo', new Route('/foo', array(), array(), array(), 'baz', array(), array(), "request.headers.get('User-Agent') matches '/firefox/i'"));
+
+        $context = new RequestContext();
+        $context->setHost('baz');
+
+        $matcher = new TraceableUrlMatcher($routes, $context);
+
+        $notMatchingRequest = Request::create('/foo', 'GET');
+        $traces = $matcher->getTracesFromRequest($notMatchingRequest);
+        $this->assertEquals("Condition \"request.headers.get('User-Agent') matches '/firefox/i'\" does not evaluate to \"true\"", $traces[0]['log']);
+
+        $matchingRequest = Request::create('/foo', 'GET', array(), array(), array(), array('HTTP_USER_AGENT' => 'Firefox'));
+        $traces = $matcher->getTracesFromRequest($matchingRequest);
+        $this->assertEquals("Route matches!", $traces[0]['log']);
```

Comment:
[fabbot](http://fabbot.io/report/symfony/symfony/17744/0969aee3117bcfc146b888df9d2b9e55d487f870) suggest to use single quote here.